### PR TITLE
VSCode Prettier

### PR DESCRIPTION
**Pull request by alder154**
Pull request for unit 5

### CODE REVIEW

File: 5 Arrays and Cycles.js [hunk 1]
```diff
@@ -1,99 +1,90 @@
-/*Дан массив имен ваших знакомых, добавить к каждому элементу массива слова hello*/
+//Дан массив имен ваших знакомых, добавить к каждому элементу массива слова hello
 
-let arr = ["Olga", "Roman", "Yana", "Oleg",];
-let result = arr.map(function(item, index, array) {
-    return ('hello ' + item);
+let arr = ["Olga", "Roman", "Yana", "Oleg"];
+let result = arr.map(function (item, index, array) {
```
Comment (rimus) [inline 5 Arrays and Cycles.js]: If you don't use paramter `array`, why do you left it in code? :)

File: 5 Arrays and Cycles.js [hunk 2]
```diff
@@ -1,99 +1,90 @@
...
+arr = str.split(",");
 console.log(arr);
 
-
 // Преобразовать числовой массив в Boolean
-let arr = [1,0,7,8,0,0,5,6];
-let result = arr.map(function(item, index, array){
-    return Boolean(item);
+let arr = [1, 0, 7, 8, 0, 0, 5, 6];
+let result = arr.map(function (item, index, array) {
```
Comment (rimus) [inline 5 Arrays and Cycles.js]: If you don't use paramter array, why do you left it in code? :)

File: 5 Arrays and Cycles.js [hunk 3]
```diff
@@ -1,99 +1,90 @@
...
-        result.push(arr[i]);
-    };
-};
+let arr = [1, 6, 7, 8, 3, 4, 5, 6];
+let result = [];
+for (let i = 0; i < arr.length; i++) {
+  if (arr[i] > 3) {
+    result.push(arr[i]);
+  }
+}
```
Comment (rimus) [inline 5 Arrays and Cycles.js]: Do you know about method `filter()`? https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

File: 5 Arrays and Cycles.js [hunk 4]
```diff
@@ -1,99 +1,90 @@
...
-            break;
-        };
-    };
-};
-console.log(callArray([8, 7, 6, 6, 5, 4, 3, 1], 6))
-
-
+function callArray(arr, n) {
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i] == n) {
```
Comment (rimus) [inline 5 Arrays and Cycles.js]: Use `===` for equation.

File: 5 Arrays and Cycles.js [hunk 5]
```diff
@@ -1,99 +1,90 @@
...
+      break;
+    }
+  }
+}
+console.log(callArray([8, 7, 6, 6, 5, 4, 3, 1], 6));
 
 //Реализовать цикл, который будет выводить число а, пока оно не станет меньше 10
 
-let a = prompt('Enter a');
+let a = prompt("Enter a");
```
Comment (rimus) [inline 5 Arrays and Cycles.js]: No prompts or alerts in your code. Please use direct pass value in your code and logging to console.

File: 5 Arrays and Cycles.js [hunk 6]
```diff
@@ -1,99 +1,90 @@
...
 while (a < 10) {
-    console.log(a);
-    a = prompt("Enter next value");
-  };
-
+  console.log(a);
+  a = prompt("Enter next value");
```
Comment (rimus) [inline 5 Arrays and Cycles.js]: No prompts or alerts in your code. Please use direct pass value in your code and logging to console.

File: 5 Arrays and Cycles.js [hunk 7]
```diff
@@ -1,99 +1,90 @@
...
 
 let a = 1;
-while (a){
-    console.log(a);
-    a = Number(prompt('press enter for next number', a+=2));
-}
\ No newline at end of file
+while (a) {
+  console.log(a);
+  a = Number(prompt("press enter for next number", (a += 2)));
```
Comment (rimus) [inline 5 Arrays and Cycles.js]: No prompts or alerts in your code. Please use direct pass value in your code and logging to console.

File: 5 Arrays and Cycles.js [hunk 8]
```diff
@@ -1,99 +1,90 @@
...
-  for (let j = 2; j < i; j++) { 
-    if (i % j == 0) continue nextPrime; 
-  };
+nextPrime: for (let i = 2; i <= n; i++) {
+  for (let j = 2; j < i; j++) {
+    if (i % j == 0) continue nextPrime;
+  }
   console.log(i);
-  i = Number(prompt("press enter for the next prime", i)); 
+  i = Number(prompt("press enter for the next prime", i));
```
Comment (rimus) [inline 5 Arrays and Cycles.js]: No prompts or alerts in your code. Please use direct pass value in your code and logging to console.

File: 5 Arrays and Cycles.js [hunk 9]
```diff
@@ -1,99 +1,90 @@
...
 console.log(1);
 let n = 3;
-nextPrime:
-for (let i = 2; i <= n; i++) {
-  for (let j = 2; j < i; j++) { 
-    if (i % j == 0) continue nextPrime; 
-  };
+nextPrime: for (let i = 2; i <= n; i++) {
+  for (let j = 2; j < i; j++) {
+    if (i % j == 0) continue nextPrime;
```
Comment (rimus) [inline 5 Arrays and Cycles.js]: Use === for equation.

File: 5 Arrays and Cycles.js [hunk 10]
```diff
@@ -1,99 +1,90 @@
...
+  }
   console.log(i);
-  i = Number(prompt("press enter for the next prime", i)); 
+  i = Number(prompt("press enter for the next prime", i));
   if (i == 1) break;
```
Comment (rimus) [inline 5 Arrays and Cycles.js]: Use === for equation.

**Review by rimus — CHANGES_REQUESTED** [review]

File: 5 Arrays and Cycles.js [hunk 11]
```diff
@@ -1,90 +1,91 @@
...
 //Реализовать цикл, который выводит в консоль простые числа
 
-console.log(1);
-let n = 3;
-nextPrime: for (let i = 2; i <= n; i++) {
-  for (let j = 2; j < i; j++) {
-    if (i % j == 0) continue nextPrime;
+function isPrimeNumber(x) {
+  if (x === 2) return x;
+  for (n = 2; n < x; n++) {
```
Comment (rimus) [inline 5 Arrays and Cycles.js]: variable `n` should be declared in loop conditions.

File: 5 Arrays and Cycles.js [hunk 12]
```diff
@@ -1,90 +1,91 @@
...
 //Реализовать цикл, который выводит в консоль нечетные числа
 
-let a = 1;
-while (a) {
-  console.log(a);
-  a = Number(prompt("press enter for next number", (a += 2)));
+function oddNumbers(num) {
+  for (let j = 1; j <= num; j += 2) {
+    console.log(j);
+  }
```
Comment (rimus) [inline 5 Arrays and Cycles.js]: negative numbers can be odd or even too. please check numbers by some condition before printing.

**Review by rimus — CHANGES_REQUESTED** [review]

File: 5 Arrays and Cycles.js (hunk 11 above)
Comment (alder154) [reply 5 Arrays and Cycles.js]: Just declared n and run this function. Either declaring or not, function works identically. Is it expected behavior?

File: 5 Arrays and Cycles.js [hunk 13]
```diff
@@ -1,99 +1,97 @@
...
+  if (num > 0) {
+    for (let j = 1; j <= num; j += 2) {
+      console.log(j);
+    }
+  } else {
+    for (let j = -1; j >= num; j -= 2) {
+      console.log(j);
+    }
+  }
+}
```
Comment (rimus) [inline 5 Arrays and Cycles.js:97]: It works but it is not a good solution. I'll show an example:
```
function printOdds(from, to) {
  console.log('Odd numbers:');
  for (let i = from; i <= to; i++) {
    if (i % 2 !== 0) {
      console.log(i);
    }
  }
}

printOdds(-15, 22);
```

Comment (rimus) [reply 5 Arrays and Cycles.js:97]: In this case we don't need to bother which numbers we are checking: positive or negative.

**Review by rimus — APPROVED** [review]

File: 5 Arrays and Cycles.js (hunk 11 above)
Comment (rimus) [reply 5 Arrays and Cycles.js]: Expected, but we always should to declare variables.

File: 5 Arrays and Cycles.js (hunk 13 above)
Comment (alder154) [reply 5 Arrays and Cycles.js:97]: Okay I see. Thank you!

**Pull request merged**